diff --git a/lib/templates/requireHooks.js b/lib/templates/requireHooks.js
index 0cfc8ac909ced4c477177fe26305da0ceb4f9029..1f5aa9437810a3b7718c6828fce9d0577d360e80 100644
--- a/lib/templates/requireHooks.js
+++ b/lib/templates/requireHooks.js
@@ -95,7 +95,7 @@ const applyRequireHooks = () => {
     ;
     module_1.default._resolveFilename = function (originalResolveFilename, hooks, request, parent, isMain, options) {
         var _a, _b;
-        const reactMode = process.env.__NEXT_PRIVATE_PREBUNDLED_REACT || 'default';
+        const reactMode = process.env.__NEXT_PRIVATE_PREBUNDLED_REACT || 'next';
         const resolvedRequest = (_b = (_a = hooks.get(reactMode)) === null || _a === void 0 ? void 0 : _a.get(request)) !== null && _b !== void 0 ? _b : request;
         return originalResolveFilename.call(module_1.default, resolvedRequest, parent, isMain, options);
         // We use `bind` here to avoid referencing outside variables to create potential memory leaks.